Automattic/sensei
# Open Student Modal when selecting bulk actions

Relate #4959

### Changes proposed in this Pull Request
* Select Courses button is implemented as React Component so it can open Student Modal Component
* I didn't touch the selector since it has lots of validation logic and lots of mentions in different PHP classes
* Unit tests added to check if the proper modal is opened when the button is clicked based on the option selected in the dropdown 

### Testing instructions
* Unit tests added to check if the proper modal is opened when the button is clicked based on the option selected in the dropdown 

*** Manual testing**

* Select **Enroll/Renroll** option in the bulk action dropdown, select students in the table and click **Select Courses** button -> Student Modal should be open with following button: 
<img width="210" alt="image" src="https://user-images.githubusercontent.com/7208249/163156505-c75bef6a-2177-44da-803f-d9f219b6fb88.png">

* Select **Remove Enrollment** option in the bulk action dropdown, select students in the table -> Student Modal should be open with following button: 
<img width="229" alt="image" src="https://user-images.githubusercontent.com/7208249/163156817-b12c24dc-bafc-4ac8-af24-0e296d3ed8ad.png">

* Select **Reset or Remove Progress** option in the bulk action dropdown, select students in the table -> Student Modal should be open with the following button: 
<img width="338" alt="image" src="https://user-images.githubusercontent.com/7208249/163156908-75e4637b-6b23-44a3-8f99-5842eb1aa2ee.png">

* Select Courses button should be disabled until you select students AND bulk action
